Upgrade log4j to 2.17 (#2890)
JanusGraph/janusgraph: docs/changelog.md

We are dropping support for old serialization format of JanusGraph predicates. The old predicates serialization format is only used by client older than 0.6. 
The change only affects Gryo and GraphSON.
##### Upgrade of log4j to version 2

> This change requires a new log4j configuration. You can find an example configuration in `conf/log4j2-server.xml`. As fact of the changed configuration format, 

Reviewer: ```suggestion
This change requires a new log4j configuration. You can find an example configuration in `conf/log4j2-server.xml`. As a result of the changed configuration format, 
```

> we cleanup all configurations. This could lead to unexpect new log lines. Please open an issue, if you have see any unwanted log line.

Reviewer: ```suggestion
we clean up all configurations. This could lead to unexpected new log lines. Please open an issue, if you see any unwanted log line.
```

!!! note 

>     Log4j is just used for standalone server deployments and JanusGraph testing.

Reviewer: ```suggestion
    Log4j is only used for standalone server deployments and JanusGraph testing.
```

We are dropping support for old serialization format of JanusGraph predicates. The old predicates serialization format is only used by client older than 0.6. 
The change only affects Gryo and GraphSON.

##### Upgrade of log4j to version 2

This change requires a new log4j configuration. You can find an example configuration in `conf/log4j2-server.xml`. As a result of the changed configuration format, 

> we clean up all configurations. This could lead to unexpected new log lines. Please open an issue, if you have see any unwanted log line.

Reviewer: nitpick: if you have see -> if you see
